pkg/cluster/internal/create/actions/config/config.go

```diff
@@ -166,7 +168,7 @@ func (a *Action) Execute(ctx *actions.ActionContext) error {
 
 // getKubeadmConfig generates the kubeadm config contents for the cluster
 // by running data through the template and applying patches as needed.
-func getKubeadmConfig(cfg *config.Cluster, data kubeadm.ConfigData, node nodes.Node) (path string, err error) {
+func getKubeadmConfig(cfg *config.Cluster, data kubeadm.ConfigData, node nodes.Node, provider string) (path string, err error) {
```

Comment:
I doubted between passing the whole context or adding a new argument, since it is only a string and doesn't look like this is going to grow much, I pick the later

Reply:
I still need to take a sledgehammer to this code anyhow :-)

this is a reasonable approach for now :+1: